### PR TITLE
Java: Models as Data inline like test.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we re-write the models as data tests to use code comments. The implementation 
- Is based on the solution for the type based model generator.
- Is made shareable between type based and dataflow model generation.
- Requires that the model comments are put on the callable from where the model originates.

### CODE REVIEW

File: java/ql/test/TestUtilities/InlineMadTest.qll [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  signature module InlineMadTestConfigSig {
+    /**
+     * Gets the kind of a captured model.
+     */
+    string getKind();
+
+    /**
+     * Gets a captured model for `c`, if any.
+     */
+    string getCapturedModel(Callable c);
```
Comment (owen-mc) [inline java/ql/test/TestUtilities/InlineMadTest.qll:27]: (trivial nitpick) We may as well put them in alphabetical order.
```suggestion
    /**
     * Gets a captured model for `c`, if any.
     */
    string getCapturedModel(Callable c);

    /**
     * Gets the kind of a captured model.
     */
    string getKind();
```

**Review by owen-mc — APPROVED** [review]
Looks great. I have only left one very trivial nitpick. Thanks for breaking it up into small commits. It was very interesting for me to see how you were structuring it to make it usable in more and more general situations. Will you move some of the code into `shared` to avoid code duplication in C#?

File: java/ql/test/TestUtilities/InlineMadTest.qll (hunk 1 above)
Comment (michaelnebel) [reply java/ql/test/TestUtilities/InlineMadTest.qll:27]: Thank you! I will do that in the follow up when moving this into the shared library!

**Comment by michaelnebel** [comment]
> Looks great. I have only left one very trivial nitpick. Thanks for breaking it up into small commits. It was very interesting for me to see how you were structuring it to make it usable in more and more general situations. Will you move some of the code into `shared` to avoid code duplication in C#?

Thank you! :-)
Yes, I will move some of the code into shared as a part of the corresponding PR for C# (I also intend to address you review comment in said PR).

**Pull request merged**